projects/igniteui-angular/migrations/common/UpdateChanges.ts

```diff
@@ -434,6 +453,182 @@ export class UpdateChanges {
+            globalPlugins: ['@angular/language-service'],
+            allowLocalPluginLoads: false,
+            typingsInstaller: ts.server.nullTypingsInstaller
+        });
+        projectService.setHostConfiguration({
+            formatOptions: projectService.getHostFormatCodeOptions(),
+            extraFileExtensions: [
+                {
+                    extension: '.html',
+                    isMixedContent: false,
+                    scriptKind: ts.ScriptKind.External,
+                }
+            ]
+        });
+        projectService.configurePlugin({
+            pluginName: '@angular/language-service',
+            configuration: {
+                angularOnly: false,
+            },
+        });
+
+        return projectService;
+    }
+
+    private updateTemplateFiles() {
+        if (this.selectorChanges && this.selectorChanges.changes.length) {
+            for (const entryPath of this.templateFiles) {
+                this.updateSelectors(entryPath);
+            }
+        }
+        if (this.outputChanges && this.outputChanges.changes.length) {
+            // name change of output
+            for (const entryPath of this.templateFiles) {
+                this.updateBindings(entryPath, this.outputChanges);
+            }
+        }
+        if (this.inputChanges && this.inputChanges.changes.length) {
+            // name change of input
+            for (const entryPath of this.templateFiles) {
+                this.updateBindings(entryPath, this.inputChanges, BindingType.input);
+            }
+        }
+    }
+
+    private updateTsFiles() {
+        if (this.classChanges && this.classChanges.changes.length) {
+            // change class name
+            for (const entryPath of this.tsFiles) {
+                this.updateClasses(entryPath);
+            }
+        }
+        if (this.importsChanges && this.importsChanges.changes.length) {
+            // TODO: move logic to 7.0.2 migration
+            for (const entryPath of this.tsFiles) {
+                this.updateImports(entryPath);
+            }
+        }
+    }
+
+    private updateMembers() {
+        if (this.membersChanges && this.membersChanges.changes.length) {
+            const dirs = [...this.templateFiles, ...this.tsFiles];
+            for (const entryPath of dirs) {
+                this.updateClassMembers(entryPath, this.membersChanges);
+            }
+        }
+    }
+
+    private findMatches(content: string, change: MemberChange): number[] {
+        let matches: RegExpExecArray;
+        const regex = this.toRegExp(change.member, 'g');
+        const matchesPositions = [];
+        do {
+            matches = regex.exec(content);
+            if (matches) {
+                matchesPositions.push(matches.index);
+            }
+        } while (matches);
+
+        return matchesPositions;
+    }
+
+    private getTypeDefinitionAtPosition(entryPath: string, position: number): ts.DefinitionInfo | null {
```

Comment:
Yeah none of those methods have much to do with the class instance here, they very well may be static methods in ts utils